### ISSUE TITLE
Help: Build guidance for arm64 M1 Mac/ update to 1.16.2 go

### ISSUE THREAD

**Issue by luke-riu** [FlowCI/flow-agent-x#45]
To try and get this running properly on an M1 Mac, I made the following changes to the makefile

```make
MAC_AMD64       := GOOS=darwin GOARCH=arm64
```

then upgraded the version of go to
```make
DOCKER_IMG              := golang:1.16.2
```
as this supports M1, however when I try to run the agent binary, it fails with 

```bash
Starting agent from binary
INFO[2021-03-28T19:49:40+01:00] Staring flow.ci agent (v0.21.05)...          
DEBU[2021-03-28T19:49:40+01:00] Port = 49646                                 
INFO[2021-03-28T19:49:40+01:00] --- [Server URL]: http://192.168.1.97:8080   
INFO[2021-03-28T19:49:40+01:00] --- [Token]: 697d3404-b595-41a7-be83-656eedc1527e 
INFO[2021-03-28T19:49:40+01:00] --- [Port]: 49646                            
INFO[2021-03-28T19:49:40+01:00] --- [Workspace]: /Users/riu-engineering/.agent.697d3404-b595-41a7-be83-656eedc1527e 
INFO[2021-03-28T19:49:40+01:00] --- [Plugin Dir]: /Users/riu-engineering/.agent.697d3404-b595-41a7-be83-656eedc1527e/.plugins 
INFO[2021-03-28T19:49:40+01:00] --- [Log Dir]: /Users/riu-engineering/.agent.697d3404-b595-41a7-be83-656eedc1527e/.logs 
INFO[2021-03-28T19:49:40+01:00] --- [Volume Str]: name=pyenv,dest=/ci/python,script=init.sh,image=flowci/pyenv:1.3,init=init-pyenv-volume.sh 
ERRO[2021-03-28T19:49:40+01:00] runtime error: invalid memory address or nil pointer dereference 
INFO[2021-03-28T19:49:40+01:00] Agent stopped 
```
any help appreciated

**Comment by luke-riu** [comment]
Further to this, I installed Golang and visual studio code on my M1 machine and had the following error when running

```bash
Failed to continue - runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation]
Unable to propogate EXC_BAD_ACCESS signal to target process and panic (see https://github.com/go-delve/delve/issues/852)
Last known immediate stacktrace (goroutine id 1):
	/Users/riu/flow-agent-x/config/manager.go:75
		github/flowci/flow-agent-x/config.(*Manager).FetchProfile
	/Users/riu/flow-agent-x/config/manager.go:104
		github/flowci/flow-agent-x/config.(*Manager).connect
	/Users/riu/flow-agent-x/config/manager.go:61
		github/flowci/flow-agent-x/config.(*Manager).Init
	/Users/riu/flow-agent-x/app.go:135
		main.start
	/Users/riu/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:490
		github.com/urfave/cli.HandleAction
	/Users/riu/go/pkg/mod/github.com/urfave/cli@v1.20.0/app.go:264
		github.com/urfave/cli.(*App).Run
	/Users/riu/flow-agent-x/app.go:91
		main.main
	/usr/local/go/src/runtime/proc.go:225
		runtime.main
	/usr/local/go/src/runtime/asm_arm64.s:1130
		runtime.goexit
	:0
		<unknown>
```

**Comment by gy2006** [comment]
I think the error came from when Agent fetching the machine memory, have you tried the latest Golang 1.16.3?

**Comment by luke-riu** [comment]
I am currently waiting for my M1 replacement machine to arrive. I will update this once I try again :)

**Comment by luke-riu** [comment]
@gy2006 I have tried this with my new Mac mini on 1.16.3 and am facing the exact same issue. Do we know if we're able to use another method of retrieving memory usage? When I last looked at this, I was able to get it working by returning mock values for memory usage and capacity so it looks like that is the only thing stopping this from running.
Thanks.

**Comment by gy2006** [comment]
I was busy moving home for the last two weeks, sorry for the late reply.
I have updated the lib of memory usage and released it with the version [v0.21.21](https://github.com/FlowCI/flow-core-x/releases/tag/v0.21.21), unfortunately, I don't have M1 Mac at moment, please help to try it if you have time, thanks a lot

**Comment by luke-riu** [comment]
this is potentially working, but I can't get it running properly as on the M1 Mac, I'm seeing the same issue as here #48
